src/coreComponents/fieldSpecification/SourceFluxStatistics.hpp

```diff
@@ -0,0 +1,420 @@
+ */
+class SourceFluxStatsAggregator : public FieldStatisticsBase< FlowSolverBase >
+{
+public:
+
+  /**
+   * @brief Aggregated flux statistics data.
+   */
+  struct StatData
+  {
+    /// Amount of fluid produced by the flux(es). Negative if injecting. One value for each fluid phase.
+    /// In kg by default, or in mol if useMass = 0 on the solver.
+    array1d< real64 > m_producedMass;
+    /// Flux(es) production rate. Negative if injecting. One value for each fluid phase.
+    /// In kg/s by default, or in mol/s if useMass = 0 on the solver.
+    array1d< real64 > m_productionRate;
+    /// Number of elements in which we are producing / injecting
+    integer m_elementCount = 0;
+
+    /**
+     * @brief resize the phase data arrays if needed
+     * @param phaseCount the count of phases
+     */
+    void allocate( integer phaseCount );
+    /**
+     * @brief reset the stats to 0.0
+     */
+    void reset();
+    /**
+     * @return the phase count for phasic stats
+     */
+    integer getPhaseCount() const
+    { return m_producedMass.size(); }
+    /**
+     * @brief Aggregate the statistics of the instance with those of another one.
+     * @param other the other stats structure.
+     */
+    void combine( StatData const & other );
+    /**
+     * @brief Aggregate the statistics of the instance with those from all instances from other MPI ranks.
+     * Must be called only once per timestep.
+     */
+    void mpiReduce();
+  };
+  /**
+   * @brief Class that aggregate statistics of a flux over multiple time-steps for a given
+   * SourceFluxStatsAggregator and a for a given mesh part (i.e. a subregion, a region...).
+   */
+  class WrappedStats
+  {
+public:
+    /**
+     * @brief Set the subjects targeted by the stats.
+     * @param aggregatorName the name of the targeted SourceFluxStatsAggregator
+     * @param fluxName the name of the targeted SourceFluxBoundaryCondition
+     */
+    void setTarget( string_view aggregatorName, string_view fluxName );
+
+    /**
+     * @brief Set the current time step stats. Accumulate the statistics only if the time is strictly
+     * after the previous time + dt (override the statistics if the current timestep is cut).
+     * @param currentTime  time of the timestep start since simulation starting
+     * @param dt           time delta of the current timestep
+     * @param producedMass Amount of produced fluid (see StatData::m_producedMass).
+     * @param elementCount number of cell elements concerned by this instance
+     */
+    void gatherTimeStepStats( real64 currentTime, real64 dt,
+                              arrayView1d< real64 const > const & producedMass,
+                              integer elementCount );
+
+    /**
+     * @brief Finalize the period statistics of each timestep gathering and render data over all mpi ranks.
+     * The result can be read by the data() assessor.
+     */
```

Comment:
Be a little more explicit about the `MPI` call that needs to be done by everyone. Maybe add a `@note` ?